File: src/main/java/com/eis/carboncredits/models/shapes/Line.java
package com.eis.carboncredits.models.shapes;

import com.fasterxml.jackson.annotation.JsonProperty;

public class Line extends Shape{
    @JsonProperty
    Point end;

    public Line(Point start, Point end) {
        super(start);
        this.end = end;
    }

    public Point getEnd() {
        return end;
    }

    public void setEnd(Point end) {
        this.end = end;
    }

    public double length(){
        double length_in_pixel;
        double x=Math.pow((this.end.x-this.start.x), 2);
        double y=Math.pow((this.end.y-this.start.y), 2);
        length_in_pixel = Math.sqrt(x+y);
        return length_in_pixel;
    }

}

File: src/main/resources/static/web/src/models/Line.ts
import { Point } from "./Point";
import { Shape } from "./Shape";

export class Line extends Shape{
    end:Point;
    constructor(start:Point, end:Point) {
        super(start)
       this.end = end; 
    }

    public length():number{
        const x=Math.pow((this.end.x-this.start.x), 2);
        const y=Math.pow((this.end.y-this.start.y), 2);
        return Math.sqrt(x+y);
    }
}